libs/execution_context/ebpf_core.c

```diff
@@ -418,6 +446,11 @@ _ebpf_core_protocol_map_update_element(_In_ const epf_operation_map_update_eleme
 
     const ebpf_map_definition_t* map_definition = ebpf_map_get_definition(map);
 
+    if (!_ebpf_is_find_update_delete_operation_permitted_on_map(map_definition)) {
+        retval = EBPF_INVALID_ARGUMENT;
+        goto Done;
+    }
+
```

Comment:
As noted above, this is not needed, since line 460 will gracefully fail in this case